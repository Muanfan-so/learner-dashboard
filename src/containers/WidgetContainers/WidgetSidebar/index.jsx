import React from 'react';

import { reduxHooks } from 'hooks';
import { PluginSlot } from '@openedx/frontend-plugin-framework';

// TODO: is this the same as NoCoursesSidebar?

// eslint-disable-next-line arrow-body-style
export const WidgetSidebar = () => {
  const hasCourses = reduxHooks.useHasCourses();

  const widgetSidebarClassNames = `widget-sidebar ${hasCourses ? '' : 'px-2'}`;
  const innerWrapperClassNames = `d-flex ${hasCourses ? 'flex-column' : ''}`;

  return (
    <div className={widgetSidebarClassNames}>
      <div className={innerWrapperClassNames}>
        <PluginSlot id="widget_sidebar_plugin_slot" />
      </div>
    </div>
  );
};

export default WidgetSidebar;
