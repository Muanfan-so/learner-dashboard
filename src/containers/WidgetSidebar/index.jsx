import React from 'react';

import LookingForChallengeWidget from './widgets/LookingForChallengeWidget';

export const WidgetSidebar = () => (
  <div className="widget-sidebar px-2 mt-5 pt-3">
    <div className="d-flex">
      <LookingForChallengeWidget />
    </div>
  </div>
);

export default WidgetSidebar;
