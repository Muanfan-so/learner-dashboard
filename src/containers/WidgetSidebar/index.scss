@import "@edx/paragon/scss/core/core";
.widget-sidebar {
  margin-top: map-get($spacers, 5);
  width: 400px;
  flex-shrink: 0;
  padding-left: map-get($spacers, 2);
  padding-right: map-get($spacers, 2);
}

@include media-breakpoint-down(md) {
  .widget-sidebar {
    display: none;
  }
}