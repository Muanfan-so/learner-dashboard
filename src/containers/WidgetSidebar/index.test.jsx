import { shallow } from 'enzyme';

import WidgetSidebar from '.';

jest.mock('./widgets/LookingForChallengeWidget', () => 'LookingForChallengeWidget');

describe('WidgetSidebar', () => {
  describe('snapshots', () => {
    test('default', () => {
      const wrapper = shallow(<WidgetSidebar />);
      expect(wrapper).toMatchSnapshot();
    });
  });
});
