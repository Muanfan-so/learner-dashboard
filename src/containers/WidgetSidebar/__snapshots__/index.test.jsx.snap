// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WidgetSidebar snapshots default 1`] = `
<div
  className="widget-sidebar px-2 mt-5 pt-3"
>
  <div
    className="d-flex"
  >
    <LookingForChallengeWidget />
  </div>
</div>
`;
